### ISSUE TITLE
inference too slow on single A100 40G

### ISSUE THREAD

**Issue by Asianfleet** [sail-sg/Consistent3D#2]
command:
python launch.py --config configs/consistency-coarse.yaml --train --gpu 0 system.prompt_processor.prompt="The Summer Palace"

output:

/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/mediapipe_face/mediapipe_face_common.py:7: UserWarning: The module 'mediapipe' is not installed. The package will have limited functionality. Please install it using the command: pip install 'mediapipe'
  warnings.warn(
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/segment_anything/modeling/tiny_vit_sam.py:654: UserWarning: Overwriting tiny_vit_5m_224 in registry with controlnet_aux.segment_anything.modeling.tiny_vit_sam.tiny_vit_5m_224. This is because the name being registered conflicts with an existing name. Please check if this is not expected.
  return register_model(fn_wrapper)
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/segment_anything/modeling/tiny_vit_sam.py:654: UserWarning: Overwriting tiny_vit_11m_224 in registry with controlnet_aux.segment_anything.modeling.tiny_vit_sam.tiny_vit_11m_224. This is because the name being registered conflicts with an existing name. Please check if this is not expected.
  return register_model(fn_wrapper)
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/segment_anything/modeling/tiny_vit_sam.py:654: UserWarning: Overwriting tiny_vit_21m_224 in registry with controlnet_aux.segment_anything.modeling.tiny_vit_sam.tiny_vit_21m_224. This is because the name being registered conflicts with an existing name. Please check if this is not expected.
  return register_model(fn_wrapper)
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/segment_anything/modeling/tiny_vit_sam.py:654: UserWarning: Overwriting tiny_vit_21m_384 in registry with controlnet_aux.segment_anything.modeling.tiny_vit_sam.tiny_vit_21m_384. This is because the name being registered conflicts with an existing name. Please check if this is not expected.
  return register_model(fn_wrapper)
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/segment_anything/modeling/tiny_vit_sam.py:654: UserWarning: Overwriting tiny_vit_21m_512 in registry with controlnet_aux.segment_anything.modeling.tiny_vit_sam.tiny_vit_21m_512. This is because the name being registered conflicts with an existing name. Please check if this is not expected.
  return register_model(fn_wrapper)
Seed set to 719
[INFO] Using 16bit Automatic Mixed Precision (AMP)
[INFO] GPU available: True (cuda), used: True
[INFO] TPU available: False, using: 0 TPU cores
[INFO] IPU available: False, using: 0 IPUs
[INFO] HPU available: False, using: 0 HPUs
[INFO] You are using a CUDA device ('GRID A100D-40C') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision
[INFO] LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
[INFO] Specify optimizer params: {'geometry.encoding': {'lr': 0.05}, 'geometry.density_network': {'lr': 0.005}, 'geometry.feature_network': {'lr': 0.005}, 'background': {'lr': 0.005}}
[INFO] 
  | Name       | Type                           | Params
--------------------------------------------------------------
0 | geometry   | ImplicitVolume                 | 12.6 M
1 | material   | DiffuseWithPointLightMaterial  | 0     
2 | background | NeuralEnvironmentMapBackground | 448   
3 | renderer   | NeRFVolumeRenderer             | 0     
--------------------------------------------------------------
12.6 M    Trainable params
0         Non-trainable params
12.6 M    Total params
50.419    Total estimated model params size (MB)
[INFO] Validation results will be saved to outputs/consistency-coarse/The_Summer_Palace/save
[INFO] Using prompt [The Summer Palace] and negative prompt []
[INFO] Using view-dependent prompts [side]:[The Summer Palace, side view] [front]:[The Summer Palace, front view] [back]:[The Summer Palace, back view] [overhead]:[The Summer Palace, overhead view]
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/mediapipe_face/mediapipe_face_common.py:7: UserWarning: The module 'mediapipe' is not installed. The package will have limited functionality. Please install it using the command: pip install 'mediapipe'
  warnings.warn(
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/segment_anything/modeling/tiny_vit_sam.py:654: UserWarning: Overwriting tiny_vit_5m_224 in registry with controlnet_aux.segment_anything.modeling.tiny_vit_sam.tiny_vit_5m_224. This is because the name being registered conflicts with an existing name. Please check if this is not expected.
  return register_model(fn_wrapper)
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/segment_anything/modeling/tiny_vit_sam.py:654: UserWarning: Overwriting tiny_vit_11m_224 in registry with controlnet_aux.segment_anything.modeling.tiny_vit_sam.tiny_vit_11m_224. This is because the name being registered conflicts with an existing name. Please check if this is not expected.
  return register_model(fn_wrapper)
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/segment_anything/modeling/tiny_vit_sam.py:654: UserWarning: Overwriting tiny_vit_21m_224 in registry with controlnet_aux.segment_anything.modeling.tiny_vit_sam.tiny_vit_21m_224. This is because the name being registered conflicts with an existing name. Please check if this is not expected.
  return register_model(fn_wrapper)
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/segment_anything/modeling/tiny_vit_sam.py:654: UserWarning: Overwriting tiny_vit_21m_384 in registry with controlnet_aux.segment_anything.modeling.tiny_vit_sam.tiny_vit_21m_384. This is because the name being registered conflicts with an existing name. Please check if this is not expected.
  return register_model(fn_wrapper)
/data/AIMH/envs/THS/lib/python3.8/site-packages/controlnet_aux/segment_anything/modeling/tiny_vit_sam.py:654: UserWarning: Overwriting tiny_vit_21m_512 in registry with controlnet_aux.segment_anything.modeling.tiny_vit_sam.tiny_vit_21m_512. This is because the name being registered conflicts with an existing name. Please check if this is not expected.
  return register_model(fn_wrapper)
tokenizer/tokenizer_config.json: 100%|██████████████████████████████████████████████████████████████████████████████████████| 807/807 [00:00<00:00, 167kB/s]
tokenizer/vocab.json: 100%|████████████████████████████████████████████████████████████████████████████████████████████| 1.06M/1.06M [00:00<00:00, 1.64MB/s]
tokenizer/merges.txt: 100%|███████████████████████████████████████████████████████████████████████████████████████████████| 525k/525k [00:00<00:00, 817kB/s]
tokenizer/special_tokens_map.json: 100%|████████████████████████████████████████████████████████████████████████████████████| 460/460 [00:00<00:00, 316kB/s]
text_encoder/config.json: 100%|█████████████████████████████████████████████████████████████████████████████████████████████| 613/613 [00:00<00:00, 139kB/s]
model.safetensors: 100%|███████████████████████████████████████████████████████████████████████████████████████████████| 1.36G/1.36G [00:47<00:00, 28.9MB/s]
[INFO] Loading Stable Diffusion ...
model_index.json: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 543/543 [00:00<00:00, 128kB/s]
scheduler/scheduler_config.json: 100%|█████████████████████████████████████████████████████████████████████████████████████| 346/346 [00:00<00:00, 80.6kB/s]
vae/config.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 553/553 [00:00<00:00, 124kB/s]
unet/config.json: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 911/911 [00:00<00:00, 211kB/s]
diffusion_pytorch_model.safetensors: 100%|███████████████████████████████████████████████████████████████████████████████| 335M/335M [00:19<00:00, 17.3MB/s]
diffusion_pytorch_model.safetensors: 100%|█████████████████████████████████████████████████████████████████████████████| 3.46G/3.46G [03:14<00:00, 17.8MB/s]
Fetching 8 files: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [03:16<00:00, 24.58s/it]
Loading pipeline components...: 100%|█████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00,  4.16it/s]
[INFO] sigma_max: 980, sigma_min: 100
[INFO] Loaded Stable Diffusion!
/data/AIMH/envs/THS/lib/python3.8/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:441: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=33` in the `DataLoader` to improve performance.
/data/AIMH/envs/THS/lib/python3.8/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:441: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=33` in the `DataLoader` to improve performance.
Epoch 0: |                                                                                                                            | 0/? [00:00<?, ?it/s]tiny-cuda-nn warning: GPUMemoryArena: GPU 0 does not support virtual memory. Falling back to regular allocations, which will be larger and can cause occasional stutter.
Epoch 0: |                                                                                                                  | 775/? [10:18Epoch 0: |                                                                                                | 931/? [12:20<00:00,  1.26it/s]Epoch 0: |                                                                                               | 1927/? [24:48<00:00,  1.29it/s]

**Comment by chengfengke** [comment]
Excuse me，I‘d to know where did you get the pre-trained parameters，if I need to download some of huggingface‘s model in advance？

**Comment by chengfengke** [comment]
哥们，我邮箱是m13521952989@163.com，能告诉我一下咋找的参数嘛，需要自己训练嘛还是什么，万分感谢
